rust/worker/src/sysdb/test_sysdb.rs

```diff
@@ -91,15 +91,8 @@ impl TestSysDb {
         if id.is_some() && id.unwrap() != segment.id {
             return false;
         }
-        if r#type.is_some() {
-            match r#type.unwrap().as_str() {
-                "hnsw" => {
-                    if segment.r#type != SegmentType::HnswDistributed {
-                        return false;
-                    }
-                }
-                _ => return false,
-            }
+        if let Some(r#type) = r#type {
```

Comment:
(the changes here are no longer necessary for this PR, but figured it'd be helpful to keep them around)

Reply:
lets keep it, thanks for cleanup